### DISCUSSION TITLE
alembic migration other tables are dropped

### DISCUSSION THREAD

**Discussion by downdawn** [fastapi-practices/fastapi_best_architecture#116] [Ideas]
During alembic migration, if other tables already exist in the database, they will be deleted

Temporary workaround:

```
def include_name(name, type_, parent_names):
    if type_ == "table":
        if re.match(r"^sys_", name):
            return False
    else:
        return True
```
```
context.configure(
        ...
        include_name=include_name,
    )
```

**Comment by wu-clan** [comment]
Should this be added as a standard? Should it be strictly required that tables not implemented in this project should not exist in the same database?

**Reply by downdawn** [reply]
It should contain all __tablename__ in the models directory

It is too inflexible to limit only one database

**Reply by wu-clan** [reply]
I don't quite understand, can you tell us what scene?

**Reply by downdawn** [reply]
Need to query the table data of other businesses, if cross-database query is too troublesome

**Reply by wu-clan** [reply]
I call it split library and split table?

**Reply by downdawn** [reply]
You can test it, create other tables in the same database, when you execute the migration, these tables will be deleted

**Reply by wu-clan** [reply]
Do you mean to create other tables in the database that are used by the business without a mod?

**Reply by downdawn** [reply]
This is a situation.

For example, what we are doing now is the background system, and the database name is sys. The business code needs to read the data of other systems without using the interface. Then there are two ways, either directly query in the same database, or create different database links.

Models can be created, but not migrated.

**Reply by wu-clan** [reply]
I don't work in development, but it seems to me that, as far as I know, calling data from other systems is called through the interfaces of other systems, not manipulating the databases of other systems

And it sounds unrealistic to directly manipulate other system's database

**Reply by downdawn** [reply]
Yes, this practice is not the norm.

But it's also unreasonable for a migration to cause the table to be dropped

**Reply by wu-clan** [reply]
I found it in the documentation, which I've been ignoring because I haven't had this additional table creation operation

https://alembic.sqlalchemy.org/en/latest/autogenerate.html#omitting-table-names-from-the-autogenerate-process

**Reply by downdawn** [reply]
Yes, I found this too. test is available
```
target_metadata = MyModel.metadata

def include_name(name, type_, parent_names):
    if type_ == "table":
        return name in target_metadata.tables
    else:
        return True

context.configure(
    # ...
    target_metadata = target_metadata,
    include_name = include_name,
    include_schemas = False
)
```

**Reply by wu-clan** [reply]
Miraculously, my local test failed😓

**Reply by downdawn** [reply]
There are two context.configure configurations, are you missing one?

**Reply by wu-clan** [reply]
Maybe you can submit this PR.

**Reply by downdawn** [reply]
#125
